app/controllers/pet_applications_controller.rb

```diff
@@ -0,0 +1,33 @@
+class PetApplicationsController < ApplicationController
+
+  def new
+    @favorite_pets = favorite.load_favorite_pets
+  end
+
+  def create
+    new_application = PetApplication.new(application_params)
+    new_application.save
+    pets = params[:pet_id]
+    if new_application.save
+      pets.each do |pet_id|
+        ApplicationPet.create(pet_id: pet_id, pet_application_id: new_application.id)
+        pet = Pet.find(pet_id)
+        pet.update(adopt_status: "Pending")
+        favorite.contents.delete(pet_id.to_i)
+      end
+      flash[:submit_app_success] = "Your application was submitted successfully!"
```

Comment:
I don't think these need to have specific "you must have an address to submit your application" style flash messages, but down the line we will need them.